src/test/java/org/zeromq/TestZThread.java

```diff
@@ -1,40 +1,30 @@
 package org.zeromq;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.notNullValue;
-import static org.junit.Assert.assertThat;
-
-import java.util.concurrent.CountDownLatch;
-
 import org.junit.Ignore;
 import org.junit.Test;
 import org.zeromq.ZMQ.Socket;
-import org.zeromq.ZThread.IAttachedRunnable;
-
 import zmq.ZError;
 
+import java.util.concurrent.CountDownLatch;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
 public class TestZThread
 {
     @Test
     public void testDetached()
     {
         final CountDownLatch stopped = new CountDownLatch(1);
-        ZThread.IDetachedRunnable detached = new ZThread.IDetachedRunnable()
-        {
-            @Override
-            public void run(Object[] args)
-            {
-                ZContext ctx = new ZContext();
-
-                Socket push = ctx.createSocket(SocketType.PUSH);
-                assertThat(push, notNullValue());
-
-                ctx.close();
-                stopped.countDown();
-            }
-        };
-
-        ZThread.start(detached);
+        ZThread.start((args) -> {
+            ZContext ctx = new ZContext();
```

Comment:
This would be a good place to use a try-with-resources block:

```java
try (ZContext ctx = new ZContext()) {
    Socket push = ctx.createSocket(SocketType.PUSH);
    assertThat(push, notNullValue());
}
```